namenode/src/main/java/org/apache/crail/namenode/TableBlocks.java

```diff
@@ -34,6 +35,10 @@ public AbstractNode putChild(AbstractNode child) throws Exception {
 			throw new Exception("Attempt to create key/value pair in container other than a table");
 		}
 		
-		return children.put(child.getComponent(), child);
+		AbstractNode oldNode = children.put(child.getComponent(), child);
+		if (child.isEnumerable()) {
+			child.setDirOffset(dirOffsetCounter.getAndAdd(CrailConstants.DIRECTORY_RECORD));
```

Comment:
Can we have a setDirOffset method in DirectoryBlocks that takes the child as an argument and sets the dir offset accordingly, e.g. something like:
```
void setChildDirOffset(AbstractNode child) {
if (child.isEnumerable()) {
	child.setDirOffset(dirOffsetCounter.getAndAdd(CrailConstants.DIRECTORY_RECORD));
}
}
```

Reply:
I'm also not a big fan of protected variable, but in this case I prefer the protected over adding another method that has a similar name and similar functionality than setDirOffset that already exists.

Reply:
I would prefer to keep it simple and keep it protected, instead of adding a new method for two reasons:

- A new method would increase code complexity
- The new method would also be inherited, either explicitly  "protected" or implicitly "package", so there would not be much of a gain, but comes at the cost of higher code complexity.

Reply:
I disagree, because if you are going to change the logic you have to remember changing it at two places. But I don't feel strong about it. I'm going to approve this.